ingest/processors/corelight.go

```diff
@@ -87,22 +89,41 @@ func (c *Corelight) Config(v interface{}, tagger Tagger) (err error) {
 	return
 }
 
+func cleanHeaders(hdrs []string) []string {
+	for i := range hdrs {
+		hdrs[i] = strings.TrimSpace(hdrs[i])
+	}
+	return hdrs
+}
+
 func (c *Corelight) init(cfg CorelightConfig, tagger Tagger) (err error) {
-	if c.Prefix == "" {
-		c.Prefix = defaultPrefix
+	var specs []corelightSpec
+	if err = cfg.Validate(); err != nil {
+		return
+	} else if specs, err = loadCustomFormats(cfg.Custom_Format); err != nil {
+		return
 	}
 	c.tagFields = make(map[string][]string, len(tagHeaders))
 	c.tags = make(map[string]entry.EntryTag)
 	var k, v string
 	for k, v = range tagHeaders {
 		tagName := c.Prefix + k
-		c.tagFields[tagName] = strings.Split(v, ",")
+		c.tagFields[tagName] = cleanHeaders(strings.Split(v, ","))
 		if tv, err := c.tg.NegotiateTag(tagName); err != nil {
 			return err
 		} else {
 			c.tags[tagName] = tv
 		}
 	}
+	for _, spec := range specs {
```

Comment:
this is almost exactly the loop directly above, could we turn it into a function call? maybe tagHeaders should be a []corelightSpec, then we could use the same set of logic

Reply:
udpated.